src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -491,25 +519,38 @@ IndexerConnector::IndexerConnector(
                 }
             }
 
-            if (!bulkData.empty())
+            // Send data to the indexer to be processed.
+            const auto processData = [&secureCommunication](const std::string& data, const std::string& url)
             {
-                auto onSuccess = [](const std::string& response)
+                const auto onSuccess = [](const std::string& response)
                 {
                     logDebug2(IC_NAME, "Response: %s", response.c_str());
                 };
 
-                auto onError = [](const std::string& error, const long statusCode)
+                const auto onError = [](const std::string& error, const long statusCode)
                 {
                     logError(IC_NAME, "%s, status code: %ld.", error.c_str(), statusCode);
                     throw std::runtime_error(error);
                 };
 
-                // Process data.
-                HTTPRequest::instance().post(RequestParameters {.url = HttpURL(url),
-                                                                .data = bulkData,
-                                                                .secureCommunication = secureCommunication},
-                                             PostRequestParameters {.onSuccess = onSuccess, .onError = onError},
-                                             {});
+                HTTPRequest::instance().post(
+                    RequestParameters {.url = HttpURL(url), .data = data, .secureCommunication = secureCommunication},
+                    PostRequestParameters {.onSuccess = onSuccess, .onError = onError},
+                    {});
+            };
+
+            const auto serverUrl = selector->getNext();
+
+            if (!bulkData.empty())
+            {
+                const auto url = serverUrl + "/_bulk?refresh=wait_for";
+                processData(bulkData, url);
+            }
+
+            if (!queryData.empty())
+            {
+                const auto url = serverUrl + "/" + m_indexName + "/_delete_by_query";
```

Comment:
This call processes multiple elements that are removed from the queue (by default it takes 1000), imagine that within those 1000 elements there are 2 delete per query. With the current logic the query would be similar to this:

```bash
curl -k -u admin:admin -X POST -d '{"query":{"match":{"agent.id":"001"}}}{"query":{"match":{"agent.id":"005"}}}' -H "Content-Type: application/json" "https://0.0.0.0:9200/wazuh-states-vulnerabilities/_delete_by_query"
```
The result of this is that only the elements of agent_id == 001 are deleted and not those of 005.


Delete by query not is a bulk query, probably queryData could be a vector(multiple http calls) or could be(single call):
```http
POST /wazuh-states-vulnerabilities/_delete_by_query
{
  "query": {
    "terms": {
      "data.agent_id": ["001", "005"]
    }
  }
}
```

Identify what is better in terms of performance, it should not be complicated to obtain these metrics.

Reply:
We made the requested changes and test the new behaviour

![image](https://github.com/user-attachments/assets/59485eea-2445-48e0-bc2e-e0f76d0b3155)

Reply:
# Aditional note

The `wait_for` only works for some actions, and not for DeleteByQuery

https://github.com/elastic/elasticsearch/issues/55970

![image](https://github.com/user-attachments/assets/c43d7054-8a0f-42e3-9da8-c4bd5ea81410)